### PR TITLE
DevicesClient and TwinsClient tests

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Microsoft.Azure.Devices.Client.csproj [hunk 1]
```diff
@@ -52,13 +52,15 @@
   <ItemGroup>
     <PackageReference Include="MQTTnet" Version="4.1.2.350" />
     <PackageReference Include="Microsoft.Azure.Amqp" Version="2.5.12" />
+    <PackageReference Include="System.Text.Json" Version="5.0.0" />
```
Comment (drwill-ms) [inline iothub/device/src/Microsoft.Azure.Devices.Client.csproj:55]: Trying out a reference to stj at v5 (first to include `[JsonInclude]` to solve the deserialization problem for internal set properties.

File: iothub/device/src/Microsoft.Azure.Devices.Client.csproj [hunk 2]
```diff
@@ -52,13 +52,15 @@
...
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
     <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0-preview1.22518.1">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
     
     <!-- Add support for IAsyncDisposable, match version used in track2 https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L80 -->
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="1.1.1" />
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="5.0.0" />
```
Comment (drwill-ms) [inline iothub/device/src/Microsoft.Azure.Devices.Client.csproj:64]: Adding v5 of STJ requires an update to this too.

Waiting for a response from Azure SDK folks for overall guidance.

File: iothub/service/src/Registry/Models/Device.cs [hunk 3]
```diff
@@ -36,6 +36,7 @@ public Device(string id)
         /// Device Id.
         /// </summary>
         [JsonPropertyName("deviceId")]
+        [JsonInclude]
```
Comment (drwill-ms) [inline iothub/service/src/Registry/Models/Device.cs:39]: Added to all properties with internal setters.

File: iothub/service/src/Twin/Models/ClientTwin.cs [hunk 4]
```diff
@@ -87,7 +87,7 @@ public ClientTwin(ClientTwinProperties twinProperties)
         /// Twin capabilities are read only.
         /// </remarks>
         [JsonPropertyName("capabilities")]
-        public ClientCapabilities Capabilities { get; internal set; }
+        public ClientCapabilities Capabilities { get; internal set; } = new();
```
Comment (drwill-ms) [inline iothub/service/src/Twin/Models/ClientTwin.cs:90]: I got an NRE in a test in deserialization, so I gave more objects default instances.

**Review by abhipsaMisra — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]

File: iothub/service/samples/getting started/JobsSample/JobsSample.cs [hunk 5]
```diff
@@ -13,7 +12,7 @@ public class JobsSample
     {
         private const string DeviceId = "JobsSample_Device";
         private const string TestTagName = "JobsSample_Tag";
-        private const int TestTagValue = 100;
+        private const long TestTagValue = 100;
```
Comment (patilsnr) [inline iothub/service/samples/getting started/JobsSample/JobsSample.cs:15]: Out of curiosity, why make this `long`? Shouldn't `int` be fine?

**Review by patilsnr — APPROVED** [review]

**Pull request merged**